### ISSUE TITLE
Installing tch-nginx-gui on Technicolor TG-1 Bricked the Gateway

### ISSUE THREAD

**Issue by fbagnato** [Ansuel/tch-nginx-gui#1171]
Hello,

I tried to install the tch-nginx-gui but the Gateway ended up in a bootloop.

I was able to reflash the firmware using TFTP and the Gateway is functional again.

I entered the commands:
curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI.tar.bz2 --output /tmp/GUI.tar.bz2
bzcat /tmp/GUI.tar.bz2 | tar -C / -xvf -
/etc/init.d/rootdevice force

The curl command seemed to work perfectly but none of the outputs from the /etc/init.d/rootdevice force seemed to work.

Is the Technicolor TG-1 supported by your tool?

Is there a way I can get it working on the Technicolor TG-1?

The attached text file is the output from the commands run.

[Ansuel_tch-nginx-gui Log.txt](https://github.com/Ansuel/tch-nginx-gui/files/12723431/Ansuel_tch-nginx-gui.Log.txt)

Regards,

fbagnato

**Comment by fbagnato** [comment]
@FrancYescO 

Can you please advise on my problem as I have set aside the router since September 26th awaiting a response on my problem.

Regards,

fbagnato

**Comment by FrancYescO** [comment]
I don't have a response for you, and I need more info about the device, maybe is a firmware v15 and pretty sure patches are needed to have it working...

If you want to retry use directly the dev version, and post a log of the install with the debug flag

**Comment by fbagnato** [comment]
Hello @FrancYescO ,

Thanks for your reply.

The device is a Technicolor TGiiNet-1 TG-1 Broadband Gateway (A TG-1 model original distributed by the Australian iiNet Internet Provider).

The firmware installed on it is version v15.53.8141-ver1.9.2.

The device is rooted.

When you say "If you want to retry use directly the dev version, and post a log of the install with the debug flag" when I ran the 1st command "curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI.tar.bz2 --output /tmp/GUI.tar.bz2" did it install the dev version?

I see that the latest version is v9.6.97. Is this the "dev" version that you speak of and if so what is the curl command to get the dev version?

Also, how do I turn the debug flag on and post a log to you?

Regards,

fbagnato

**Comment by fbagnato** [comment]
Hello @FrancYescO ,

I'm still waiting on your reply from my last post.

Could you please attend to it?

Regards,

fbagnato

**Comment by fbagnato** [comment]
Hello @FrancYescO ,

I'm still waiting on your reply from my last post.

I still haven't been able to complete the modification on my router and would appreciate your reply.

Could you please attend to it?

Regards,

fbagnato

**Comment by FrancYescO** [comment]
'curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /tmp/GUI_dev.tar.bz2
bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf -
/etc/init.d/rootdevice debug'

**Comment by fbagnato** [comment]
Hello @FrancYescO

I tried the command you provided but got an error. Is the syntax correct????

Here is the output:

root@dsldevice:~# curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice debug
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3460k  100 3460k    0     0  1627k      0  0:00:02  0:00:02 --:--:-- 1631k
curl: (6) Couldn't resolve host 'bzcat'
curl: (3) <url> malformed
tar: short read

**Comment by FrancYescO** [comment]
i updated the command up, seems there was a missing `|`

**Comment by fbagnato** [comment]
Thanks for your prompt reply @FrancYescO

Now when I run the command most likely it will re-brick the router as it has in the past.

Where does the debug command put the debug file you need? If it is on the router I won't be able to access it as the router will be bricked.

Regards,

Flavio

**Comment by FrancYescO** [comment]
it will just give more logs during the install, and pretty sure will go in the same "bricked" status.

as i assument you already done, you can restore the router via TFTP/reset button.

**Comment by fbagnato** [comment]
Nope, got the following:

root@dsldevice:~# curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 | bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice debug
bzcat: /tmp/GUI_dev.tar.bz2: No such file or directory
tar: short read
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3460k  100 3460k    0     0  1389k      0  0:00:02  0:00:02 --:--:-- 1406k

**Comment by fbagnato** [comment]
Can I just run the official command on the main page of:

curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI.tar.bz2 --output /tmp/GUI.tar.bz2
bzcat /tmp/GUI.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice force debug

Just adding debug to the end of the command?

Regards,

Flavio

**Comment by FrancYescO** [comment]
you should also replace all the three occurence of GUI.tar.bz2 with GUI_dev.tar.bz2

**Comment by fbagnato** [comment]
Hello @FrancYescO,

I got the following error:

curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /tmp/GUI_dev.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0 3460k    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0Warning: Failed to open the file /tmp/GUI_dev.tar.bz2: No such file or
Warning: directory
  2 3460k    2 98304    0     0  23249      0  0:02:32  0:00:04  0:02:28 24135
curl: (23) Failure writing output to destination

C:\WINDOWS\system32>bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice force debug

**Comment by fbagnato** [comment]
C:\WINDOWS\system32>curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /tmp/GUI_dev.tar.bz2 bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice force debug
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to open the file /tmp/GUI_dev.tar.bz2: No such file or
Warning: directory
  0 3460k    0 16384    0     0  56855      0  0:01:02 --:--:--  0:01:02 87148
curl: (23) Failure writing output to destination
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0curl: (6) Could not resolve host: bzcat
curl: (3) URL rejected: No host part in the URL
tar: /etc/init.d/rootdevice: Not found in archive
tar: force: Not found in archive
tar: debug: Not found in archive
tar: Error exit delayed from previous errors.

**Comment by FrancYescO** [comment]
1) you are using a windows shell and not the router shell
2) your command is still wrong and missing a pipe after the curl command

**Comment by fbagnato** [comment]
Hello @FrancYescO 

I forgot to SSH into the router on the last command run.

Can you please provide the correct command to run.

My last attempt was the command:
curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 | bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice force debug

Output was:
root@dsldevice:~# curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 | bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf - /etc/init.d/rootdevice force debug
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3460k  100 3460k    0     0  1132k      0  0:00:03  0:00:03 --:--:-- 1135k
tar: /etc/init.d/rootdevice: not found in archive

**Comment by fbagnato** [comment]
Hello @FrancYescO

Should there be a new line between two commands?

ie. should the command be:

curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 | bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf -
*********************NEWLINE*****************************
 /etc/init.d/rootdevice force debug

Regards,

Flavio

**Comment by fbagnato** [comment]
Hello @FrancYescO

Can you please help me, I am not an expert and am limited as to what I can figure out myself.

Regards,

Flavio

**Comment by FrancYescO** [comment]
> Should there be a new line between two commands?\

yes

**Comment by fbagnato** [comment]
Hello @FrancYescO 

I don't understand what I am doing wrong!

Here is the 1st command I ran:

curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 | bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf -

Here is the output from the first command:

root@dsldevice:~# curl -k https://raw.githubusercontent.com/Ansuel/gui-dev-build-auto/master/GUI_dev.tar.bz2 --output /t
mp/GUI_dev.tar.bz2 | bzcat /tmp/GUI_dev.tar.bz2 | tar -C / -xvf -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0bzcat: bunzip error -3
tar: short read
100 3460k  100 3460k    0     0  1606k      0  0:00:02  0:00:02 --:--:-- 1611k

Can you please provide the correct command to run?

Regards,

Flavio

**Comment by fbagnato** [comment]
@FrancYescO

Do you know why the last command failed with a bzcat: bunzip error -3
tar: short read?

Does this mean there is something wrong with the GUI_dev.tar.bz2 file?

What command should I run?

Can I replace GUI_dev.tar.bz2 with GUI.tar.bz2?

I hope you can reply so that we can finally get the debug command to successfully run..

Regards,

Flavio

**Comment by fbagnato** [comment]
Hello @FrancYescO

Finally a positive result, the commands you provided ran successfully.

Thank-you for persisting with me.

I have attached the output from the commands.
[Ansuel_tch-nginx-gui Debug Log.txt](https://github.com/Ansuel/tch-nginx-gui/files/14450979/Ansuel_tch-nginx-gui.Debug.Log.txt)

Please let me know what happens next. 

Is it possible to install tch-nginx-gui on this router?

Regards,

Flavio

**Comment by fbagnato** [comment]
Hello @FrancYescO

Just wondering if you had a chance to look at my submitted debug log?

Regards,

Flavio

**Comment by fbagnato** [comment]
Hello @FrancYescO

Just wondering if you had a chance to look at my submitted debug log?

I'm still waiting on your input so are you able to reply?

Regards,

Flavio

**Comment by fbagnato** [comment]
Hello @FrancYescO

Just wondering if you had a chance to look at my submitted debug log?

I'm still waiting on your input so are you able to reply?

Regards,

fbagnato

**Comment by FrancYescO** [comment]
Unfortunately not, is not going to happen soon

**Comment by fbagnato** [comment]
Hello @FrancYescO,

Thanks for the update, much appreciated.

I would like to know if this router can be updated with tch-nginx-gui so I would appreciate an answer whenever you get a chance.

Regards,

fbagnato

**Comment by fbagnato** [comment]
Hello @FrancYescO,

My problem is still outstanding.

I would like to know if this router can be updated with tch-nginx-gui so I would appreciate an answer whenever you get a chance.

Regards,

fbagnato

**Comment by FrancYescO** [comment]
Not until someone fix the code to manage errors you have

**Comment by fbagnato** [comment]
But aren't you the author of the tch-nginx-gui code?

What is the reason none of the outputs from the /etc/init.d/rootdevice force seem to work?

I have previously provided the debug output for your analysis.
